scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -545,9 +546,16 @@
       a bearer token for authenticating with kubernetes
     - bearer-token-refresh-seconds: interval to refresh the bearer token"
   [^String config-file base-path ^String use-google-service-account? bearer-token-refresh-seconds verifying-ssl ^String ssl-cert-path]
-  (let [api-client (if (some? config-file)
-                     (Config/fromConfig config-file)
-                     (ApiClient.))
+  (log/info "API Client config file" config-file)
+  (let [^ApiClient api-client (if (some? config-file)
+                                (let [^KubeConfig kubeconfig
+                                      (-> config-file
+                                          (FileInputStream.)
+                                          (InputStreamReader. (.name (StandardCharsets/UTF_8)))
+                                          (KubeConfig/loadKubeConfig))]
+                                  (.setFile kubeconfig (File. config-file)) ; Workaround client library bug. Needed to avoid a NPE. Use an absolute path so this isn't used.
```

Comment:
The comment isn't clear to me -- what was the bug / NPE?

Reply:
Rewrote